### ISSUE TITLE
SAI Challenger not catching create/remove failures?

### ISSUE THREAD

**Issue by chrispsommers** [opencomputeproject/SAI-Challenger#30]
Running the following test appears to pass but in fact there are errors upon create and remove which the framework does not catch:
https://github.com/Azure/DASH/blob/main/test/test-cases/scale/saic/test_config_vnet_inbound.py

Complete details are here: [saichallenger-vnet-inbound-rounting-errors.txt](https://github.com/opencomputeproject/SAI-Challenger/files/10310367/saichallenger-vnet-inbound-rounting-errors.txt)

The following configuration JSON snippet is applied to the bmv2 DUT via `process_commands()` and the bmv2 console and saithrift console clearly show failures. The failure appears to be ignored by the SAIC framework for create. For remove, it does return `False`.
```
  {
    "name": "inbound_routing_entry",
    "op": "create",
    "type": "SAI_OBJECT_TYPE_INBOUND_ROUTING_ENTRY",
    "key": {
      "switch_id": "$SWITCH_ID",
      "eni_id": "$eni_id",
      "vni": "1000",
      "sip": "10.10.2.0",
      "sip_mask": "255.255.255.0",
      "priority": 0
    },

    "attributes": [
      "SAI_INBOUND_ROUTING_ENTRY_ATTR_ACTION",
      "SAI_INBOUND_ROUTING_ENTRY_ACTION_VXLAN_DECAP_PA_VALIDATE",
      "SAI_INBOUND_ROUTING_ENTRY_ATTR_SRC_VNET_ID",
      "$vnet_1"
    ]
  },
```
To summarize:
1. There may be an underlying bmv2 error which fails on the create op
2. SAIC does not report any error for failed create
3. SAIC returns False for failed remove op

**Comment by chrispsommers** [comment]
@andriy-kokhan Him any updates on this? Thanks.

**Comment by andriy-kokhan** [comment]
Let me double check how SAI-C processes the errors returned by Thrift and get back to you.

**Comment by chrispsommers** [comment]
> Let me double check how SAI-C processes the errors returned by Thrift and get back to you.

OK, but it seems like this is a sai-challenger issue - why does its behavior change when a program supplies both `key` and a `name`? I'd think this could be caught in the client itself: detect the presence of both and reject the entry before transforming it into an AIP call (assuming that is the appropriate action to take).

**Comment by andriy-kokhan** [comment]
Expected to be fixed by https://github.com/opencomputeproject/SAI-Challenger/commit/3f6e90c5f5283782abcc21261861d369752144f2

**Comment by andriy-kokhan** [comment]
I believe this issue has already been addressed. Please confirm.

**Comment by chrispsommers** [comment]
@andriy-kokhan Thanks, I just re-tested using today's DASH main branch (commit SHA = `00dde8cc146e4a68ce28da815a8194b497990153`) and it appears the test is correctly catching the create error. 

I temporarily commented out a [skipped test](https://github.com/sonic-net/DASH/blob/00dde8cc146e4a68ce28da815a8194b497990153/test/test-cases/scale/saic/test_config_vnet_inbound.py#L11) to expose the error-handling. Then I ran one test as follows:
```
SAI_CHALLENGER_TEST=test_config_vnet_inbound.py make run-saichallenger-scale-tests
```
Failure was properly caught and reported:
```
>           assert status == 'SAI_STATUS_SUCCESS', f"create({obj}, {attrs}) --> {status}"
E           AssertionError: create(SAI_OBJECT_TYPE_VIP_ENTRY:{"switch_id": "oid:0x0", "vip": "192.168.0.1"}, ['SAI_VIP_ENTRY_ATTR_ACTION', 'SAI_VIP_ENTRY_ACTION_ACCEPT']) --> SAI_STATUS_FAILURE
```
Closing issue, thanks again!
